### DISCUSSION TITLE
Publishing flow-typed definitions to npm

### DISCUSSION THREAD

**Discussion by Brianzchen** [flow-typed/flow-typed#4366] [General]
In the past there has been many discussions in regards to shift definitions to be installed via npm instead of a dedicated `flow-typed` cli. 

Although the idea is a possibility it requires a complete overhaul of the CLI and re-architecture of how we manage definitions. There's no clarity still on how to manage library definitions that span across multiple flow versions as [explained](https://github.com/flow-typed/flow-typed/issues/990#issuecomment-313609695).

We've since created the ability for definitions to depend on one another which was one of the driving forces behind distributing through npm.

## tldr;

This serves as the mega thread for anyone who still wants to shift definition distribution through npm over a custom CLI and would like to discuss their ideas with the community as opposed to several issues which are hard for newcomers to find.
